site-src/references/policy-attachment.md

```diff
@@ -303,3 +510,241 @@ the same behavior and semantics, although they may not be able to support
+defaulting and overriding of the `retryOn` field.
+
+A full `RetryOnPolicy` to default the field to the codes `501`, `502`, and `503`
+would look like this:
+```yaml
+apiVersion: networking.example.io/v1alpha1
+kind: RetryOnPolicy
+metadata:
+  name: retryon5xx
+  namespace: appns
+spec:
+  defaults:
+    retryOn:
+      - "501"
+      - "502"
+      - "503"
+  targetRef:
+    kind: Gateway
+    group: gateway.networking.k8s.io
+    name: we-love-retries
+```
+
+This means that, for HTTPRoutes that do _NOT_ explicitly set this field to something
+else, (in other words, they contain an empty list), then the field will be set to
+a list containing `501`, `502`, and `503`. (Notably, because of Go zero values, this
+would also occur if the user explicitly set the value to the empty list.)
+
+However, if a HTTPRoute owner sets any value other than the empty list, then that
+value will remain, and the Policy will have _no effect_. These values are _not_
+merged.
+
+If the Policy used `overrides` instead:
+```yaml
+apiVersion: networking.example.io/v1alpha1
+kind: RetryOnPolicy
+metadata:
+  name: retryon5xx
+  namespace: appns
+spec:
+  overrides:
+    retryOn:
+      - "501"
+      - "502"
+      - "503"
+  targetRef:
+    kind: Gateway
+    group: gateway.networking.k8s.io
+    name: you-must-retry
+```
+
+Then no matter what the value is in the HTTPRoute, it will be set to `501`, `502`,
+`503` by the Policy override.
+
+### Interactions between defaults, overrides, and field values
+
+All HTTPRoutes that attach to the `YouMustRetry` Gateway will have any value
+_overwritten_ by this policy. The empty list, or any number of values, will all
+be replaced with `501`, `502`, and `503`.
+
+Now, let's also assume that we use the Namespace -> Gateway hierarchy on top of
+the Gateway -> HTTPRoute hierarchy, and allow attaching a `RetryOnPolicy` to a
+_namespace_. The expectation here is that this will affect all Gateways in a namespace
+and all HTTPRoutes that attach to those Gateways. (Note that the HTTPRoutes
+themselves may not necessarily be in the same namespace though.)
+
+If we apply the default policy from earlier to the namespace:
+```yaml
+apiVersion: networking.example.io/v1alpha1
+kind: RetryOnPolicy
+metadata:
+  name: retryon5xx
+  namespace: appns
+spec:
+  defaults:
+    retryOn:
+      - "501"
+      - "502"
+      - "503"
+  targetRef:
+    kind: Namespace
+    group: ""
+    name: appns
+```
+
+Then this will have the same effect as applying that Policy to every Gateway in
+the `default` namespace - namely that every HTTPRoute that attaches to every
+Gateway will have its `retryOn` field set to `501`, `502`, `503`, _if_ there is no
+other setting in the HTTPRoute itself.
+
+With two layers in the hierarchy, we have a more complicated set of interactions
+possible.
+
+Let's look at some tables for a particular HTTPRoute, assuming that it does _not_
+configure the `retryOn` field, for various types of Policy at different levels.
```

Comment:
Sigh. I guess you're right. I always anticipated that it would apply to fields, but it make sense for it not to. I'll update and remove the references to `retryOn` being a field.

Reply:
If you feel really strongly about it, I can change it, but that will involve quite a bit of change to the example (to use a Filter instead or something).

Reply:
Up to this point I've thought of policy attachment as purely an extension mechanism, not something that actually overrides behavior specified by the resource. I think right now the UX would be pretty awful if this actually happened. It's one thing to say "some other resource may add config on top of this route to extend it", but it's an entirely different thing to say "some other resource may completely override the config you've specified in this route." If we want policy attachment to be able to solve the latter I think we need more than just status on the policy resource, but actually on the target resource. 

It's not obvious to me that we _need_ this today, so I'd rather leave it out of scope until we do because I'm quite concerned about the UX issues this would introduce. The lack of status is bad enough when we're extending functionality, it would be even more pronounced if we were overriding functionality specified by the API. Of course I can see the usefulness of a capability like this, but so far the best use cases seem to be concepts we haven't included in the API (timeouts, retries, etc), and when we include them, they may just be policies.

Reply:
Yeah, okay. I'll update the example to be a RetryFilter that's controlled by a RetryPolicy.

Reply:
I've updated. I actually think it makes this a bit harder to understand, but I get that we don't want to imply you can do something we don't know if we want to allow.